pkg/storage/fs/ocis/revisions.go

```diff
@@ -0,0 +1,45 @@
+package ocis
+
+import (
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	"github.com/cs3org/reva/pkg/errtypes"
+)
+
+func (fs *ocisfs) ListRevisions(ctx context.Context, ref *provider.Reference) (revisions []*provider.FileVersion, err error) {
+	var node *NodeInfo
+	if node, err = fs.pw.NodeFromResource(ctx, ref); err != nil {
+		return
+	}
+	if !node.Exists {
+		err = errtypes.NotFound(filepath.Join(node.ParentID, node.Name))
+		return
+	}
+
+	revisions = []*provider.FileVersion{}
+	nodePath := filepath.Join(fs.pw.Root(), "nodes", node.ID)
+	if items, err := filepath.Glob(nodePath + ".REV.*"); err == nil {
```

Comment:
what if nodePath is the string "abc.REV.xx.yy.txt" ? better move revisions into a separate directory

Reply:
the entries inside the node folder all start with a uuid, `.REV.` indicates it is a revision. and what follows is a timestamp, so multiple versions can be kept in the same location as the current actual file content. This prevents new fileuploads to trigger cross storage moves when revisions accidentially are stored on another partition, because they admin mounted a different partition there. We can make a background process move old revisions to a slower storage and replace the revision five with a symbolic link in the future if necessary.